operators/src/plot/histogram.rs

```diff
@@ -0,0 +1,841 @@
+            process_metadata(processor.query(query, ctx), &self.column_name, self.metadata)
+        })
+    }
+
+    async fn process<'p>(
+        &'p self,
+        metadata: HistogramMetadata,
+        query: QueryRectangle,
+        ctx: &'p dyn QueryContext,
+    ) -> Result<<HistogramRasterQueryProcessor as PlotQueryProcessor>::PlotType> {
+        let mut histogram = geoengine_datatypes::plots::Histogram::builder(
+            metadata.number_of_buckets,
+            metadata.min,
+            metadata.max,
+            self.measurement.clone(),
+        )
+        .build()
+        .map_err(Error::from)?;
+
+        call_on_generic_vector_processor!(&self.input, processor => {
+            let mut query = processor.query(query, ctx);
+
+            while let Some(collection) = query.next().await {
+                let collection = collection?;
+
+                let feature_data = collection.data(&self.column_name).expect("checked in param");
+
+                histogram.add_feature_data(feature_data)?;
+            }
+        });
+
+        let chart = histogram.to_vega_embeddable(self.interactive)?;
+
+        Ok(chart)
+    }
+}
+
+#[derive(Debug, Copy, Clone, PartialEq)]
+struct HistogramMetadata {
+    pub number_of_buckets: usize,
+    pub min: f64,
+    pub max: f64,
+}
+
+#[derive(Debug, Copy, Clone, PartialEq)]
+struct HistogramMetadataOptions {
+    pub number_of_buckets: Option<usize>,
+    pub min: Option<f64>,
+    pub max: Option<f64>,
+}
+
+impl TryFrom<HistogramMetadataOptions> for HistogramMetadata {
+    type Error = ();
+
+    fn try_from(options: HistogramMetadataOptions) -> Result<Self, Self::Error> {
+        match (options.number_of_buckets, options.min, options.max) {
+            (Some(number_of_buckets), Some(min), Some(max)) => Ok(Self {
+                number_of_buckets,
+                min,
+                max,
+            }),
+            _ => Err(()),
+        }
+    }
+}
+
+impl HistogramMetadataOptions {
+    fn merge_with(self, metadata: HistogramMetadata) -> HistogramMetadata {
+        HistogramMetadata {
+            number_of_buckets: self.number_of_buckets.unwrap_or(metadata.number_of_buckets),
+            min: self.min.unwrap_or(metadata.min),
+            max: self.max.unwrap_or(metadata.max),
+        }
+    }
+}
+
+#[derive(Debug, Copy, Clone, PartialEq)]
+struct HistogramMetadataInProgress {
+    pub n: usize,
+    pub min: f64,
+    pub max: f64,
+}
+
+impl Default for HistogramMetadataInProgress {
+    fn default() -> Self {
+        Self {
+            n: 0,
+            min: f64::MAX,
+            max: f64::MIN,
+        }
+    }
+}
+
+impl HistogramMetadataInProgress {
+    #[inline]
+    fn add_raster_batch<T: Pixel>(&mut self, values: &[T], no_data: Option<T>) {
+        if let Some(no_data) = no_data {
+            for &v in values {
+                if v == no_data {
+                    continue;
+                }
+
+                self.n += 1;
```

Comment:
an optimization would be to compute min/max and bucket number separately, so if users specifies bucket count but not min/max you don't compute bucket count in vein. Maybe something for moonshots.

Reply:
fd3417b 

I guess there are lots of improvements, also for the actual calculation.